src/controllers/krate/publish.rs

```diff
@@ -450,22 +449,45 @@ pub async fn publish(app: AppState, req: Parts, body: Body) -> AppResult<Json<Go
         // Compared to only using a background job, this prevents us from getting into a
         // situation where a crate exists in the `crates` table but doesn't have a default
         // version in the `default_versions` table.
-        if let Some(existing_default_version) = existing_default_version {
+        if let Some((existing_default_version, existing_total)) = existing_default_version {
             let published_default_version = DefaultVersion {
                 id: version.id,
                 num: semver,
                 yanked: false,
             };
-
-            if existing_default_version < published_default_version {
-                diesel::update(default_versions::table)
-                    .filter(default_versions::crate_id.eq(krate.id))
-                    .set(default_versions::version_id.eq(version.id))
+            let update_base = diesel::update(default_versions::table)
+                    .filter(default_versions::crate_id.eq(krate.id));
+
+            // If `total` isn't set (it defaults to 0), this only occurs when a crate already
+            // exists and releases a new version before the admin sets the total.
+            // In this scenario, the `total` value is updated in a background job. This case should
+            // be safe to remove once all `total` values have been set.
+            let is_new = existing_default_version < published_default_version;
+            let default_total = existing_total == 0;
```

Comment:
wouldn't it be easier to have the column nullable for now and work with `None/Some(..)` instead?